tests/validation/recver_val.c

```diff
@@ -0,0 +1,359 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 Intel Corporation
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <bsd/string.h>
+#include <getopt.h>
+#include <signal.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <time.h>
+#include <unistd.h>
+#include "mesh_dp.h"
+#include "common.c"
```

Comment:
Why did you decided to include .c file instead of building it and linking?
Common defines are usually in .h file.

Reply:
Pure laziness. I had to remind myself how to do that properly. Hopefully, how I have done it is the right way... ;)

Plus, found a few minor bugs, which I fixed as well in the meantime, so it was definitely worth it.